Assets/LeapMotion/Modules/HierarchyRecording/Scripts/LeapRecording/VectorHandRecording/QuaternionCurves.cs

```diff
@@ -60,11 +60,17 @@ public struct QuaternionCurves {
     }
 
     public Quaternion Evaluate(float time) {
-      Quaternion evaluated = new Quaternion(xCurve.Evaluate(time),
-                                            yCurve.Evaluate(time),
-                                            zCurve.Evaluate(time),
-                                            wCurve.Evaluate(time));
-      return Quaternion.Lerp(evaluated, evaluated, 1.0f);
+      var x = xCurve.Evaluate(time);
+      var y = yCurve.Evaluate(time);
+      var z = zCurve.Evaluate(time);
+      var w = wCurve.Evaluate(time);
+      if (x == 0f && y == 0f && z == 0f && w == 0f) {
```

Comment:
Would want to use something like Mathf.Approximately to compare these, because there are non-zero values that can still trip up Quaternion.Lerp because they are so small.

Reply:
Sure, I can add that, although the case was primarily meant to catch `default(Quaternion)` specifically.

Reply:
In fact instead of trying to match the exact behavior of Quaternion.Lerp, it might just be safer to calculate the magnitude of the quaternion manually, and do the normalization yourself.  That way you can be sure you never pass a quaternion to Lerp that Lerp is unable to normalize.

Reply:
Oh, well, upon some basic research that appears to be very straight-forward. TIL Quaternions are literally just four-dimensional vectors and normalizing them means exactly what you would expect. I'll do that instead.